core/state_transition.go

```diff
@@ -174,8 +175,8 @@ func (st *StateTransition) to() common.Address {
 
 func (st *StateTransition) buyGas() error {
 	mgval := new(big.Int).Mul(new(big.Int).SetUint64(st.msg.Gas()), st.gasPrice)
-	if st.state.GetBalance(st.msg.From()).Cmp(mgval) < 0 {
-		return ErrInsufficientFunds
+	if have, want := st.state.GetBalance(st.msg.From()), mgval; have.Cmp(want) < 0 {
+		return fmt.Errorf("%w: address 0x%x have %v want %v", ErrInsufficientFunds, st.msg.From(), have, want)
```

Comment:
Perhaps you can use `fmt.Errorf("%s", st.msg.From().Hex())`, but yeah it's basically identical. Please feel to ignore this comment

Reply:
Oh, I did your suggestion, and it's actually more correct, because then it also prints it with checksummed casing, which is neat